### ISSUE TITLE
incorrect memory management in portable export format

### ISSUE THREAD

**Issue by Smattr** [polyml/polyml#89]
`PExport::exportStore` contains the following logic:

```cpp
if (nObjects == nMapSize)
{
    // Need to expand the array.
    PolyObject **newMap =
        (PolyObject **)realloc(pMap, (nMapSize + nMapSize/2)*sizeof(PolyObject*));
    if (newMap == 0)
        throw MemoryException();
    pMap = newMap;
}
```

I think this block should also be updating `nMapSize`. Otherwise, we can only expand this buffer once and then each future expansion simply reallocates a buffer of the same size. I haven't actually attempted to trigger this behaviour, this is simply an observation from reading the code so I may have this wrong.

I would be inclined to just rephrase this function to use a `std::vector<PolyObject*>` and let libstdc++ handle growing this array as needed.

**Comment by dcjm** [comment]
You're quite right and I've committed a fix (0e902c29).  The code has developed over a long time and while std::vector is used in a few places there are still a lot of places where realloc is used, many dating back from the time when the RTS was written in C rather than C++.

**Comment by Smattr** [comment]
Thanks for the quick fix, David! It looks like that commit in question could have also dropped `PExport::totalBytes` which looks no longer used, as well as the following block of logic in `PExport::exportStore` which no longer seems to have any effect:

```cpp
// Calculate a first guess for the map size based on the space size
totalBytes = 0;
void *startAddr = 0;
for (i = 0; i < memTableEntries; i++)
{
    totalBytes += (unsigned long)memTable[i].mtLength;
    // Get the lowest address.
    if (startAddr == 0 || memTable[i].mtAddr < startAddr)
        startAddr = memTable[i].mtAddr;
}
```
